Source/missiles.cpp

```diff
@@ -2606,7 +2606,7 @@ void MI_LArrow(int i)
 			if (missile[i]._mix != missile[i]._miVar1 || missile[i]._miy != missile[i]._miVar2) {
 				missile[i]._miVar1 = missile[i]._mix;
 				missile[i]._miVar2 = missile[i]._miy;
-				ChangeLight(missile[i]._mlid, missile[i]._mix, missile[i]._miy, 5);
+				ChangeLight(missile[i]._mlid, missile[i]._miVar1, missile[i]._miVar2, 5);

```

Comment:
Does this mean that it was incorrect in vanilla? Prior to changing the field `_mix` to `_miVar1`?

Reply:
No, it was just poor decompilation due to optimizing. It is assigned to the same variables right above it i.e. `missile[i]._miVar1 = missile[i]._mix;`. So likely `_mix` is put in a register and then pushed afterwards for minimum size.

Reply:
Right. This rings a bell somehow. Thanks for explaining it to me (perhaps for the second time, seeing as I have a deja vu feeling).

Reply:
There have been a few PRs like this one today 🙂